File: codebuild_specs/aws-configure-credentials.py
import os

AWS_ACCESS_KEY_ID = os.environ['AWS_ACCESS_KEY_ID']
AWS_SECRET_ACCESS_KEY = os.environ['AWS_SECRET_ACCESS_KEY']
AWS_SESSION_TOKEN = os.environ['AWS_SESSION_TOKEN']

credentials_file_contents = '''[default]
aws_access_key_id={AWS_ACCESS_KEY_ID}
aws_secret_access_key={AWS_SECRET_ACCESS_KEY}
aws_session_token={AWS_SESSION_TOKEN}

'''

config_file_contents = '''[default]
region=us-west-2

'''

with open('/root/.aws/credentials', 'w') as f:
  f.write(credentials_file_contents)

with open('/root/.aws/config', 'w') as f:
  f.write(config_file_contents)

File: scripts/aws-configure-credentials.ts
import { join } from 'path';
import * as fs from 'fs';

const AWS_ACCESS_KEY_ID = process.env['AWS_ACCESS_KEY_ID'];
const AWS_SECRET_ACCESS_KEY = process.env['AWS_SECRET_ACCESS_KEY'];
const AWS_SESSION_TOKEN = process.env['AWS_SESSION_TOKEN'];

const credentials_file_contents = `[default]
aws_access_key_id=${AWS_ACCESS_KEY_ID}
aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
aws_session_token=${AWS_SESSION_TOKEN}

`;

const config_file_contents = `[default]
region=us-west-2

`;

fs.writeFileSync(join('/', 'root', '.aws', 'credentials'), credentials_file_contents);
fs.writeFileSync(join('/', 'root', '.aws', 'config'), config_file_contents);
